xarray/core/dataset.py

```diff
@@ -2324,11 +2356,6 @@ def _calculate_binary_op(self, f, other, join='inner',
                              inplace=False, fillna=False):
 
         def apply_over_both(lhs_data_vars, rhs_data_vars, lhs_vars, rhs_vars):
-            if fillna and join != 'left':
```

Comment:
you can remove the `fillna` argument from the function signature entirely (and also from `Dataset._binary_op`)